Framework/Algorithms/src/CalculateDIFC.cpp

```diff
@@ -71,11 +71,15 @@ void CalculateDIFC::exec() {
   DataObjects::OffsetsWorkspace_sptr offsetsWs =
       getProperty("OffsetsWorkspace");
   API::MatrixWorkspace_sptr inputWs = getProperty("InputWorkspace");
-  API::MatrixWorkspace_sptr outputWs =
-      boost::dynamic_pointer_cast<MatrixWorkspace>(SpecialWorkspace2D_sptr(
-          new SpecialWorkspace2D(inputWs->getInstrument())));
-
-  outputWs->setTitle("DIFC workspace");
+  API::MatrixWorkspace_sptr outputWs = getProperty("OutputWorkspace");
+
+  if ((!bool(inputWs == outputWs)) ||
+      (!bool(boost::dynamic_pointer_cast<SpecialWorkspace2D>(outputWs)))) {
+    outputWs =
+        boost::dynamic_pointer_cast<MatrixWorkspace>(SpecialWorkspace2D_sptr(
```

Comment:
`SpecialWorkspace2D_sptr(new SpecialWorkspace2D(inputWs->getInstrument())` should be changed to `boost::make_shared<SpecialWorkspace2D>(inputWs->getInstrument())`